ecosystem/indexer-grpc/indexer-grpc-fullnode/src/stream_coordinator.rs

```diff
@@ -98,12 +98,22 @@ impl IndexerStreamCoordinator {
                 // Fetch and convert transactions from API
                 let raw_txns =
                     Self::fetch_raw_txns_with_retries(context.clone(), ledger_version, batch).await;
+                let first_raw_transaction = raw_txns.first().unwrap();
+                let last_raw_transaction = raw_txns.last().unwrap();
+                let mut last_transaction_timestamp = None;
+                if enable_expensive_logging {
+                    let api_txn = Self::convert_to_api_txns(context.clone(), vec![
+                        last_raw_transaction.clone(),
+                    ])
+                    .await;
+                    let pb_txn = Self::convert_to_pb_txns(api_txn);
+                    last_transaction_timestamp = pb_txn.first().unwrap().timestamp.clone();
```

Comment:
Should we just do the first and last timestamp?

Reply:
this is only the last transaction timestamp. i'm reusing the api and pb conversion methods here which need a vec